torch/csrc/jit/script/compiler.cpp

```diff
@@ -2767,17 +2767,15 @@ struct to_ir {
         if (type_hint && type_hint->kind() == TypeKind::ListType) {
           elem_type = type_hint->expect<ListType>()->getElementType();
```

Comment:
This isn't a result of your PR, but maybe while you're here could you fix the weird behavior where if the type hint isn't a list we ignore it ? (error if it's not a list)